file-adapter/src/main/java/org/polypheny/db/adapter/file/Condition.java

```diff
@@ -175,6 +175,41 @@ Object getParamValue( final DataContext dataContext, final PolyType polyType ) {
         return out;
     }
 
+    /**
+     * Implement the like keyword
+     *
+     * @param str Data in database
+     * @param expr String in SQL statement
+     * @return
+     */
+    private static boolean like( final String str, String expr ) {
+        final String[] parts = expr.split( "%" );
+        final boolean traillingOp = expr.endsWith( "%" );
+        expr = "";
+        for ( int i = 0, l = parts.length; i < l; ++i ) {
+            final String[] p = parts[i].split( "\\\\\\?" );
+            if ( p.length > 1 ) {
+                for ( int y = 0, l2 = p.length; y < l2; ++y ) {
+                    expr += p[y];
+                    if ( i + 1 < l2 ) {
+                        expr += ".";
+                    }
+                }
+            } else {
+                expr += parts[i];
+            }
+            if ( i + 1 < l ) {
+                expr += "%";
+            }
+        }
+        if ( traillingOp ) {
+            expr += "%";
+        }
+        expr = expr.replace( "_", "." );
+        expr = expr.replace( "%", ".*" );
+
+        return str.matches( expr );
```

Comment:
As a small improvement for faster processing you could initially check if the passed expression does even contain wildcards or any operators that you need to work on.
If not you could directly execute the:  `return str.matches( expr );`

E.g If you have any queries like:

*  `select * from dummy where text like 'Han.'`
*  `select * from dummy where text like 'Hans'` (Or even queries that do not really need the like whatsoever, but its still specified)

This would speed up the query performance for rather 'ill-defined' `like` operations.

Reply:
@erdengk 
Thank you for your review, I will submit it again after I revise it.